lua/nvim-tree/view.lua

```diff
@@ -66,17 +66,60 @@ M.View = {
   }
 }
 
+---Find a rogue NvimTree buffer that might have been spawned by i.e. a session.
+---@return integer|nil
+local function find_rogue_buffer()
+  for i = 1, vim.fn.bufnr("$"), 1 do
+    if vim.fn.bufname(i) == "NvimTree" then
+      return i
+    end
+  end
+  return nil
+end
+
+---Check if the tree buffer is valid. The buffer is only valid if it exists and
+---hasn't been deleted or wiped.
+---@return boolean
+local function is_buf_valid()
+  local success, valid = pcall(function ()
+    return (
+      a.nvim_buf_is_valid(M.View.bufnr)
+      and a.nvim_buf_get_var(M.View.bufnr, "nvim_tree_buffer_ready") == 1
+      )
+  end)
+
+  if not success then
+    return false
+  else
+    return valid
+  end
+end
```

Comment:
i think you could just return `a.nvim_buf_is_valid(M.View.bufnr) and a.nvim_buf_is_loaded(M.View.bufnr)`

Reply:
Ah, right. That's better. I'm too used to the vim script way.